src/large_pages/node_large_page.cc

```diff
@@ -251,6 +263,40 @@ struct text_region FindNodeTextRegion() {
       size = 0;
     }
   }
+#elif defined(__sun)
+  auto size = 1 << 20;
+  auto fd = open("/proc/self/map", O_RDONLY);
+
+  if (fd != -1) {
+    ssize_t r;
+    prmap_t* cur, * map;
+    std::unique_ptr<char[]> pm = std::unique_ptr<char[]>(new char[size]);
+
+    while ((r = pread(fd, pm.get(), size, 0)) == size) {
```

Comment:
Not a c/c++ savvy but is a manual `open` + `pread` + manual_char_array necessary here, could we use something from C++ (like `ifstream`, `seekg` etc) and then reinterpret cast?

Reply:
I used C here as realised on solaris/illumos C I/O are faster than C++ ans the C usage is very little. However does not mean the code here can t improve over time.

Reply:
It's okay then, thanks.